crates/scarb-api/src/lib.rs

```diff
@@ -282,19 +288,88 @@ mod tests {
             .run()
             .unwrap();
 
-        let result = try_get_starknet_artifacts_path(
+        let path = get_starknet_artifacts_path(
             &Utf8PathBuf::from_path_buf(temp.to_path_buf().join("target")).unwrap(),
             "basic_package",
             "dev",
-        );
-        let path = result.unwrap().unwrap();
+            false,
+        )
+        .unwrap();
+
         assert_eq!(
             path,
             temp.path()
                 .join("target/dev/basic_package.starknet_artifacts.json")
         );
     }
 
+    #[test]
+    #[cfg_attr(not(feature = "scarb_2_8_3"), ignore)]
+    fn get_starknet_artifacts_path_for_test_build() {
+        let temp = setup_package("basic_package");
+
+        ScarbCommand::new_with_stdio()
+            .current_dir(temp.path())
+            .arg("build")
+            .arg("--test")
+            .run()
+            .unwrap();
+
+        let path = get_starknet_artifacts_path(
+            &Utf8PathBuf::from_path_buf(temp.to_path_buf().join("target")).unwrap(),
+            "basic_package",
+            "dev",
+            true,
+        )
+        .unwrap();
+
+        assert_eq!(
+            path,
+            temp.path()
+                .join("target/dev/basic_package_unittest.test.starknet_artifacts.json")
+        );
+    }
+
+    #[test]
+    #[cfg_attr(not(feature = "scarb_2_8_3"), ignore)]
+    fn get_starknet_artifacts_path_for_test_build_with_tests_dir() {
+        let temp = setup_package("basic_package");
+        let tests_dir = temp.join("tests");
+        fs::create_dir(&tests_dir).unwrap();
+
+        temp.child(tests_dir.join("test.cairo"))
+            .write_str(indoc!(
+                r"
+                #[test]
+                fn mock_test() {
+                    assert!(true);
+                }
+            "
+            ))
+            .unwrap();
+
+        ScarbCommand::new_with_stdio()
+            .current_dir(temp.path())
+            .arg("build")
+            .arg("--test")
+            .run()
+            .unwrap();
+
+        let path = get_starknet_artifacts_path(
+            &Utf8PathBuf::from_path_buf(temp.to_path_buf().join("target")).unwrap(),
+            "basic_package",
+            "dev",
+            true,
+        )
+        .unwrap();
+
+        assert_eq!(
+            path,
+            temp.path()
+                .join("target/dev/basic_package_integrationtest.test.starknet_artifacts.json")
+        );
+    }
```

Comment:
Should we also assert that unittests are still there?

Reply:
We can, but this test is specifically to check that when we have both contract artifacts for `unittest` and `integrationtest` then it should load `integrationtest`

Reply:
Okay, this is not 100% clear that this is the purpose of the test. Perhaps rename or restructure it in a way that it indicates that better.

Reply:
Renamed to `get_starknet_artifacts_path_for_test_build_when_integration_tests_exist`